### DISCUSSION TITLE
v2.8

### DISCUSSION THREAD

**Discussion by Ghost-chu** [PBH-BTN/PeerBanHelper#40] [公告]
## 重要错误修复

* 修复了 Peer 在重新连接到下载器时，如果此前连接过但未完成数据传输断开并重连时，因下载器提供了数据（但因未完成交换数据和握手）而导致数据不完整的情况下，进度检查其出现错误封禁的问题

## 重要规则更改 

* 默认规则配置文件新增 `hp/torrent`, `-HPXXXX-`
  * 旧的规则配置文件不会自动升级，如果您是从旧版升级，请手动添加它们（或者删除 `profile.yml` 使其重新生成）

### IP段连锁封禁

批量部署的恶意客户端通常在同一个 IP 段下，PBH 现在允许用户分别为 IPv4 和 IPv6 设置一个前缀长度。在封禁发现的吸血客户端时，会将其所处 IP 地址的指定范围的其余 IP 地址均加入屏蔽列表，实现链式封禁。

（图片仅供演示，默认配置文件不会封禁整个 /24 而是更合理的 /30）

旧的规则配置文件不包含此模块的配置项，请从 Github 复制对应配置段，或者删除 `profile.yml` 使其重新生成。我们正在对配置文件系统进行改造，完成后将支持配置文件升级功能。 

![Snipaste_2024-04-11_00-12-43](https://github.com/Ghost-chu/PeerBanHelper/assets/30802565/fa564264-2e39-42d0-b8fd-2222fc1e5bc5)

![Snipaste_2024-04-11_00-12-35](https://github.com/Ghost-chu/PeerBanHelper/assets/30802565/8928eca3-83b3-4826-9727-a605cbaedcd3)

### 封禁历史

在 PBH 引入嵌入式数据库后，现在有能力持久记录封禁的历史记录。你可以在这里找到目前和过去被封禁的客户端的具体情况。

![Snipaste_2024-04-11_00-08-18](https://github.com/Ghost-chu/PeerBanHelper/assets/30802565/7e5b020a-4275-497a-9522-fc89bf90d473)

### 简易分析

提供了分析最常被封禁的 IP 的功能

![Snipaste_2024-04-11_00-08-33](https://github.com/Ghost-chu/PeerBanHelper/assets/30802565/7a323ffc-453c-440b-a54c-4f58230b2cef)

### 错误修复

* 修复默认配置文件的特殊符号错误 (@BUnipendix ) (首次贡献)
* 收到 SIGTERM 信号量后，现在 PBH 能够安全退出（@Gaojianli ) 

<hr /><em>This discussion was created from the release <a href='https://github.com/Ghost-chu/PeerBanHelper/releases/tag/v2.8'>v2.8</a>.</em>

**Comment by Mishasama** [comment]
可以考虑在日志和排名里增加`添加到内置的永久封禁名单`的功能。
